File: mediq-backend/src/main/java/com/mediq/service/NurseService.java
package com.mediq.service;

import com.mediq.dto.PatientVitalsResponse;
import com.mediq.dto.RecordVitalsRequest;

import java.util.List;

public interface NurseService {

    PatientVitalsResponse recordVitals(RecordVitalsRequest request);

    PatientVitalsResponse getVitalsById(Long vitalsId);

    PatientVitalsResponse getVitalsByTokenId(Long queueTokenId);

    List<PatientVitalsResponse> getPatientVitalsHistory(Long patientId);
}

File: mediq-frontend/src/services/nurseService.js
import api from './api';

export const nurseService = {
  recordVitals: async (data) => api.post('/nurse/vitals', data),
  getVitalsById: async (id) => api.get(`/nurse/vitals/${id}`),
  getVitalsByTokenId: async (tokenId) => api.get(`/nurse/vitals/token/${tokenId}`),
  getPatientVitalsHistory: async (patientId) => api.get(`/nurse/vitals/patient/${patientId}`),
};

export default nurseService;
